docs/references/keywords.rst

```diff
@@ -194,7 +194,46 @@ extensions).
 .. _keyword/package_dir:
 
 ``package_dir``
-    A dictionary providing a mapping of package to directory names.
+    A dictionary that maps package names (as the developer wishes they would be
+    imported by the end-users) into directory paths (that actually exist in the
+    project's source tree). This configuration can be used with 2 main objectives:
+
+    1. To effectively "rename" paths when building your package.
+       For example, ``package_dir={"mypkg": "dir1/dir2/code_for_mypkg"}``
+       will instruct setuptools to copy the ``dir1/dir2/code_for_mypkg/...`` files
+       as ``mypkg/...`` when building the final :term:`wheel distribution <Wheel>`.
+
+       .. attention::
+          While it is *possible* to specify arbitrary mappings, developers are
+          **STRONGLY ADVISED AGAINST** that. They should try as much as possible
+          to keep the directory names and structure identical to as they would
+          appear in the final wheel, only deviating when absolutely necessary.
+
+    2. To indicate that the code corresponding to a package implementation is
+       entirely contained inside a specific directory.
```

Comment:
```suggestion
    2. To indicate that the relevant code is entirely contained inside
       a specific directory (instead of directly placed under the project's root).
```